content/dart/concepts/type-conversion/type-conversion.md

```diff
@@ -0,0 +1,496 @@
+Uncaught Error: RangeError (fractionDigits): Invalid value: Not in inclusive range 0..20: 21
+```
+
+---
+
+## toStringAsExponential() Method
+
+## toStringAsExponential Implementation
+
+```pseudo
+String toStringAsExponential([int? fractionDigits]);
+```
+
+The `toStringAsExponential` method converts the input number to a `double` before computing the exponential string representation of that number.
+
+If the parameter `fractionDigits` is given, then it must be an integer satisfying:
+
+  0 <= `fractionDigits` <= 20.
+
+Meaning the `fractionDigits` must be a Non-Zero and a Positive value and must be less than 21.
+
+Otherwise, without the parameter, the returned string uses the shortest number of digits that accurately represent this number.
+
+If `fractionDigits` equals 0, then the decimal point is omitted.
+
+## Example
+
+```dart
+void main() {
+
+  var a = 1.toStringAsExponential(3);
+  var b = (256463.51554546).toStringAsExponential(3);
+  var c = (2351.44464661).toStringAsExponential();
+  var d = 254645466111.toStringAsExponential(3);
+  var e = 1224.toStringAsExponential(20);
+  var f = 5.25.toStringAsExponential(0);
+
+  print(a);
+  print(b);
+  print(c);
+  print(d);
+  print(e);
+  print(f);
+
+}
+```
+
+The `Shell` output:
+
+```Shell
+1.000e+0
+2.565e+5
+2.35144464661e+3
+2.546e+11
+1.22400000000000000000e+3
+5e+0
+```
+
+---
+
+## toStringAsPrecision() Method
+
+## toStringAsPrecision Implementation
+
+```pseudo
+String toStringAsPrecision(int precision);
+```
+
+The `toStringAsPrecision` method converts the input number to a `double` and returns a string representation of that value with exactly `precision` significant digits.
+
+The parameter `precision` must be an integer satisfying:
+  
+  1 <= `precision` <= 21.
+
+## Example1
+
+```dart
+void main() {
+
+  var a = 1.toStringAsPrecision(3);
+  var b = (256463.51554546).toStringAsPrecision(1);
+  var c = (2351.44464661).toStringAsPrecision(2);
+  var d = 254645466111.toStringAsPrecision(3);
+  var e = 1224.toStringAsPrecision(20);
+  var f = 5.25.toStringAsPrecision(12);
+
+  print(a);
+  print(b);
+  print(c);
+  print(d);
+  print(e);
+  print(f);
+
+}
+```
+
+The `Shell` output:
+
+```Shell
+1.00
+3e+5
+2.4e+3
+2.55e+11
+1224.0000000000000000
+5.25000000000
+```
+
+---
+
+## toDouble() Method
+
+The `toDouble()` method converts a value to a double.
+
+## toDouble Method Example
+
+```dart
+void main() {
+  int intValue = 42;
+  double doubleValue = intValue.toDouble();
+
+  var checkCondition = (doubleValue == 42.0) ? true : false;
+
+  print(checkCondition);
+}
+```
+
+The `Shell` output:
+
+```Shell
+true
+```
+
+---
+
+## toInt() Method
+
+The `toInt()` method converts a value to an integer.
+
+## toInt Method Example
+
+```dart
+void main() {
+  double doubleValue = 42.75;
+  int intValue = doubleValue.toInt();
+
+  var checkCondition = (intValue == 42) ? true : false;
+
+  print(checkCondition);
+}
+```
+
+The `Shell` output:
+
+```Shell
+true
+```
+
+---
+
```

Comment:
We don't need any specific function explanation or example in this case. Please remove the above-mentioned lines.

Reply:
Done.